daal4py/sklearn/_utils.py

```diff
@@ -176,6 +178,23 @@ def get_number_of_types(dataframe):
         return 1
 
 
+def check_tree_nodes(tree_nodes):
```

Comment:
Did you checked perf with this new checker?

Reply:
Perf ratio for `sklearnex.ensemble` and `sklearnex.preview.ensemble` is nearly 1, so perf check is passed